selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/FindReferencesConsoleTab.java

```diff
@@ -52,25 +64,37 @@ private boolean isReferenceContainsText(WebElement reference, String expectedTex
     return seleniumWebDriverHelper.waitVisibilityAndGetText(reference).contains(expectedText);
   }
 
-  public void waitReferenceWithText(String expectedText) {
+  private void waitReferenceWithText(String expectedText) {
     seleniumWebDriverHelper.waitSuccessCondition(
         driver ->
             getReferences()
                 .stream()
                 .anyMatch(reference -> isReferenceContainsText(reference, expectedText)));
   }
 
+  public void waitAllReferencesWithText(String... expectedText) {
+    asList(expectedText).forEach(text -> waitReferenceWithText(text));
+  }
+
   public void clickOnReference(String visibleText) {
     for (WebElement reference : getReferences()) {
       if (isReferenceContainsText(reference, visibleText)) {
         seleniumWebDriverHelper.waitAndClick(reference);
+        return;
       }
     }
 
     String errorMessage = format("No reference with text \"%s\" has been detected", visibleText);
     throw new RuntimeException(errorMessage);
   }
 
+  public void doubleClickOnReference(String visibleText) {
+    Actions action = actionsFactory.createAction(seleniumWebDriver);
```

Comment:
seleniumWebDriverHelper.getAction()

Reply:
even more. You can remove this row and replace the last row of the method ( ```action.doubleClick().perform();``` ) by:
```
seleniumWebDriverHelper.doubleClick();
```